app/templates/_widget.ts

```diff
@@ -0,0 +1,42 @@
+/// <reference path="../../../tsd.ts.ts" />
+/// <amd-dependency path="dojo/text!./templates/<%=widgetName%>.html" name="template" />
+declare var template: any;
+
+//dojo
+import dojoDeclare = require("dojo/_base/declare");
+
+//dijit
+import WidgetBase = require("dijit/_WidgetBase");
+import TemplatedMixin = require("dijit/_TemplatedMixin"); 
+<% if(widgetsInTemplate) { %>import WidgetsInTemplateMixin = require("dijit/_WidgetsInTemplateMixin"); <% }%>
+
+interface WidgetInterface {
```

Comment:
I understand. That's an interesting approach. 

> the type created by dojoDeclare is type any

Do you mean when using the official Dojo type definitions?

I don't use TS day to day, so I don't have a better suggestion. My hope would be that using the official Dojo/Esri type definitions would make it so you don't have to do this kind of thing. Personally, I'd rather not have the intellisense in this case than to have to write and maintain my own interface in multiple locations which could (however unlikely) get out of sync w/ the actual API.